packages/view-util/lib/projection.js

```diff
@@ -0,0 +1,60 @@
+import * as ns from '@view-builder/core/ns.js'
+import { rdf, schema } from '@tpluscode/rdf-ns-builders'
+
+/**
+ * Iterated view dimensions and return a set of columns to set as `view:projection/view:columns`.
+ *
+ * For every dimension whose values are IRIs, yields a new label lookup dimension.
+ * Otherwise, yields the dimension itself.
+ */
+export async function getColumns(view, cubeLookup) {
+  const dimensions = await Promise.all(view.out(ns.view.dimension)
+    .map(async dimension => ({
+      dimension,
+      isIri: await cubeLookup.isIriDimension(dimension),
+    })))
+
+  return (function * iterate() {
+    let i = 0
+    function nextColumnProperty() {
+      i += 1
+      return view.namedNode(`${view.value}#column${i}`)
+    }
+
+    for (const { dimension, isIri } of dimensions) {
+      dimension.addOut(ns.view.as, nextColumnProperty())
+      yield dimension
+
+      if (isIri) {
+        const labelDimension = view.blankNode()
+
+        labelDimension
+          .addOut(ns.view.as, nextColumnProperty())
+          .addOut(ns.view.labelFor, dimension)
```

Comment:
I don't know if `view:labelFor` is the best name; I wonder if `lookupTarget` would be more precise.